modules/reconciliation/model/reconciliation.class.php

```diff
@@ -0,0 +1,1120 @@
+      global $_lib;
+
+      $VoucherResult = $_lib['db']->db_query($VoucherQuery);
+      $AmountInH  = array();
+      $AmountOutH = array();
+      while ($Voucher = $_lib['db']->db_fetch_object($VoucherResult)) {
+        if ($Voucher->AmountIn != 0) {
+          $AmountInH[$Voucher->VoucherID]  += $Voucher->AmountIn;
+        }
+        if ($Voucher->AmountOut != 0) {
+          $AmountOutH[$Voucher->VoucherID] += $Voucher->AmountOut;
+        }
+        $Balance += ($Voucher->AmountIn - $Voucher->AmountOut);
+      }
+
+      // We have to round the amount because of imprecision in PHP calculations
+      $Balance = round($Balance, 3);
+      return array($Balance, $AmountInH, $AmountOutH);
+    }
+
+    /**
+     * Run SQL to close/reconcile two vouchers
+     */
+    private function closePostSQL($VoucherID1, $VoucherID2) {
+      global $_lib;
+
+      $VoucherQuery = "
+        SELECT
+          *
+        FROM
+          voucher
+        WHERE
+          VoucherID IN ($VoucherID1, $VoucherID2) AND
+          VoucherReconciliationID IS NOT NULL";
+      $VoucherResult = $_lib['storage']->db_query($VoucherQuery);
+      if ($_lib['storage']->db_numrows($VoucherResult) < 2) {
+        if ($_lib['storage']->db_numrows($VoucherResult) == 0) {
+          $ReconciliationInsertQuery = "
+            INSERT INTO
+              voucherreconciliation(CreatedAt, CreatedBy)
+            VALUES
+              (NOW(), " . $_lib['sess']->get_person('PersonID') . ")";
+          $ReconciliationID = $_lib['db']->db_insert($ReconciliationInsertQuery);
+        } else {
+          $ReconciliationIDQuery = "
+            SELECT
+              VoucherReconciliationID
+            FROM
+              voucher
+            WHERE
+              VoucherID IN ($VoucherID1, $VoucherID2) AND
+              VoucherReconciliationID IS NOT NULL";
+          $ReconciliationIDRow = $_lib['storage']->get_row(array('query' => $ReconciliationIDQuery));
+          $ReconciliationID = $ReconciliationIDRow->VoucherReconciliationID;
+        }
+        $UpdateVoucherReconciliationIDQuery = "
+          UPDATE
+            voucher
+          SET
+            VoucherReconciliationID = $ReconciliationID
+          WHERE
+            VoucherID IN ($VoucherID1, $VoucherID2)";
+        $_lib['db']->db_update($UpdateVoucherReconciliationIDQuery);
+      } else {
+        // $_lib['message']->add("Duplikatlukking: $VoucherID1:$VoucherID2");
+      }
+    }
+
+    /**
+     * Unreconciles all vouchers, in all periods for all account plans
+     */
+    function openAllPosts() {
+      global $_lib;
+
+      $DeleteReconciliationsQuery = "DELETE voucherreconciliation";
+      $_lib['db']->db_delete($DeleteReconciliationsQuery);
+      $UpdateVouchersQuery = "
+        UPDATE
+          voucher
+        SET
+          v.VoucherReconciliationID = NULL
+        WHERE
+          v.Active = 1";
+      $_lib['db']->db_update($UpdateVouchersQuery);
+    }
+
+    /**
```

Comment:
For all account plans, we do not have this type of action. Delete? Discuss with @stoivo